pliers/utils/base.py

```diff
@@ -117,3 +118,7 @@ def env_keys(cls):
     @classproperty
     def available(cls):
         return all([k in os.environ for k in cls.env_keys])
+
+    @abstractmethod
+    def validate_keys(self):
+        pass
```

Comment:
Not implementing this method does not seem to raise an error, as expected with `abstractmethod`s.
I believe that this is a problem since `EnvironmentKeyMixin` is typically the second parent class. Not a big problem and not sure how to address it.

Reply:
Are you setting the metaclass to `ABCMeta`? I think that's required for these decorators to work properly. (Also, since I've gotten burned by this myself recently, note that we'll have to use the `with_metaclass()` function in `six` in order to support this behavior on both Python 2 and 3).